### PR TITLE
influxdb: include build commit and time

### LINKED ISSUE #78867: InfluxDB build doesn't inject commit for display by `/health` API and UI
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config
HOMEBREW_VERSION: 3.1.9-133-gf1f7dc1
ORIGIN: https://github.com/Homebrew/brew
HEAD: f1f7dc151dc0c188f32e08ddd4d31240abb94804
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: cbdbf68578a6e80dcce5feba73c8862aac1a833d
Core tap last commit: 9 minutes ago
Core tap branch: dm-influxdb-set-commit-info
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 12.0.5 build 1205
Git: 2.32.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.4-x86_64
CLT: 12.5.0.22.9
Xcode: N/A

$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/dan/go/bin/pkg-config
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Test that the auto-upgrade of `influxdb` to 2.0.7 properly handled the UI, since the UI-bundling was broken in 2.0.6.

### What happened (include all command output)?

The build/install link works fine, but once running the server I noticed that the `/health` endpoint wasn't returning all the information I'd expect:
```bash
# With `INFLUXD_CONFIG_PATH=/usr/local/etc/influxdb2/config.yml influxd` running in the background
$ curl localhost:8086/health
{"name":"influxdb", "message":"ready for queries and writes", "status":"pass", "checks":[], "version": "2.0.7", "commit": "none"}
```
The `"commit": "none"` is the surprising piece.

This translates to a strange thing in the UI landing page:
Screen Shot 2021-06-07 at 9 50 30 AM

### What did you expect to happen?

I expected the short-hash of the `influxdb` commit to be injected at build time so it can be returned via the API and displayed in the UI.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install influxdb
brew services start influxdb
curl localhost:8086/health
```

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by SMillerDev — APPROVED** [review]

File: Formula/influxdb.rb [hunk 1]
```diff
@@ -54,9 +56,18 @@ def install
...
     # Build the CLI and server.
-    ldflags = "-s -w -X main.version=#{version}"
-    system "go", "build", *std_go_args(ldflags: ldflags), "-o", bin/"influx", "./cmd/influx"
-    system "go", "build", *std_go_args(ldflags: ldflags), "-tags", "assets", "-o", bin/"influxd", "./cmd/influxd"
+    ldflags = %W[
+      -s
+      -w
+      -X main.version=#{version}
+      -X main.commit=#{Utils.git_short_head(length: 10)}
+      -X main.date=#{Time.now.utc.iso8601}
```
Comment (bayandin) [inline Formula/influxdb.rb:64]: Won't this produce unreproducible builds? 
Shall we use `ENV["SOURCE_DATE_EPOCH"]` here?

Comment (SMillerDev) [reply Formula/influxdb.rb:64]: Sounds good to me!

Comment (alebcay) [reply Formula/influxdb.rb:64]: Yes, but upstream doesn't have reproducible builds either. Just sticking with upstream here. If it's really a concern feel free to open a PR to fix; to me, I don't see much of a benefit if upstream isn't even on board with it.

Comment (alebcay) [reply Formula/influxdb.rb:64]: I've done some more reading up on what you suggested, @bayandin. Will open a PR for what you propose when I have a chance.

Comment (bayandin) [reply Formula/influxdb.rb:64]: Nice! Thanks for the PR @alebcay!

**Pull request closed without merging**